File: Graph/ScriptTemplate/DegToRadian/CGDegToRadian.lua
local CGDegToRadian = CGDegToRadian or {}
CGDegToRadian.__index = CGDegToRadian

function CGDegToRadian.new()
    local self = setmetatable({}, CGDegToRadian)
    self.inputs = {}
    self.outputs = {}
    self.nexts = {}
    return self
end

function CGDegToRadian:setInput(index, func)
    self.inputs[index] = func
end

function CGDegToRadian:getOutput(index)
    if self.inputs[0]() then
        return math.rad(self.inputs[0]())
    end
    return nil
end

return CGDegToRadian

File: Graph/ScriptTemplate/DegToRadian/CGDegToRadian.js
/**
 * @file CGDegToRadian.js
 * @author xuyuan
 * @date 2021/8/28
 * @brief CGDegToRadian.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const Amaz = effect.Amaz;
const {BaseNode} = require('./BaseNode');

class CGDegToRadian extends BaseNode {
  constructor() {
    super();
  }
  getOutput(index) {
    return this.inputs[0]() * (Math.PI / 180);
  }
}

exports.CGDegToRadian = CGDegToRadian;
